File: LeetCode/Easy/0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        if(ransomNote.length() > magazine.length()){
            return false;
        }
        // 1. 创建一个长度为 26 的整型数组，用于统计字符频率
        // 默认初始值都是 0
        int[] count = new int[26];
        for(char c : magazine.toCharArray()){
            // c - 'a' 计算出字符所对应的索引(0-25)
            // ++ 表示将该位置的计数值 + 1
            count[c - 'a']++;

        }
        for(char c : ransomNote.toCharArray()){
            count[c -'a']--;
            if(count[c - 'a'] < 0){
                return false;
            }
        }
        return true;

    }
}

File: LeetCode/Easy/0383-ransom-note/0383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if len(ransomNote) > len(magazine):
            return False
        map = {}
        for c in magazine:
            map[c] = map.get(c, 0) + 1
        for c in ransomNote:
            map[c] = map.get(c, 0) - 1
        for v in map.values():
            if v < 0:
                return False
        return True
        